src/store.c

```diff
@@ -339,112 +375,190 @@ static void writer_close(
     close(store->fd);
 }
 
+static void init_store_offsets(
```

Comment:
I would include `init_store_offsets` in `writer_open` as you need to call the former after every call to the later which means that requiring 2 calls instead increases the error surface.